currency/code_types.go

```diff
@@ -41,12 +41,11 @@ type Code struct {
 // Item defines a sub type containing the main attributes of a designated
 // currency code pointer
 type Item struct {
-	ID            int      `json:"id"`
-	FullName      string   `json:"fullName"`
-	Symbol        string   `json:"symbol"`
-	Role          Role     `json:"role"`
-	AssocChain    string   `json:"associatedBlockchain"`
-	AssocExchange []string `json:"associatedExchanges"`
+	ID         int    `json:"id,omitempty"`
```

Comment:
Is omitempty useful on this? int's have a default value of 0 so would this ever be empty since its not a pointer?

Also seems like fullname symbol role are required and should never be empty